app/pages/nonsipp/landorpropertydisposal/WhenWasPropertySoldPage.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages.nonsipp.landorpropertydisposal
+
+import config.Refined.{Max50, Max5000}
+import models.SchemeId.Srn
+import pages.QuestionPage
+import play.api.libs.json.JsPath
+import utils.RefinedUtils.RefinedIntOps
+
+import java.time.LocalDate
+
+case class WhenWasPropertySoldPage(srn: Srn, landOrPropertyIndex: Max5000, disposalIndex: Max50)
+    extends QuestionPage[LocalDate] {
+
+  override def path: JsPath =
+    Paths.disposalPropertyTransaction \ toString \ "dateOfSale"
+
+  override def toString: String = "propertyDateOfSale"
```

Comment:
Needs the right path, disposalIndex in path and toString values. Also to update the mapping document.